presto-client/src/main/java/com/facebook/presto/client/QueryResults.java

```diff
@@ -203,12 +204,15 @@ public String toString()
             return null;
         }
         requireNonNull(columns, "columns is null");
+        List<TypeSignature> signatures = columns.stream()
+                .map(column -> parseTypeSignature(column.getType()))
+                .collect(toList());
```

Comment:
I simply used toList instead of toImmutableList. Because `presto-main` is not in the presto-client's pom. I can change that if requested.